src/Cli/dotnet/Program.cs

```diff
@@ -139,9 +139,14 @@ internal static int ProcessArgs(string[] args, TimeSpan startupTime, ITelemetry
                             ToolPathSentinelFileName)));
                 if (parseResult.ValueForOption<bool>(Parser.DiagOption) && parseResult.IsDotnetBuiltInCommand())
                 {
-                    Environment.SetEnvironmentVariable(CommandContext.Variables.Verbose, bool.TrueString);
-                    CommandContext.SetVerbose(true);
-                    Reporter.Reset();
+                    // We found --diagnostic or -d, but we still need to determine whether the option should
+                    // be attached to the dotnet command or the subcommand.
+                    if (args.DiagOptionPrecedesSubcommand(parseResult.RootSubCommandResult()))
```

Comment:
So, it seems that sub-commands also can have aliases. So, for this to not cause a bug if a sub-command alias is created, you'd need to check both the sub-command name and its aliases. Maybe just jam the name and aliases into a collection and pass the whole collection in. Then, add a `foreach` to the `DiagOptionPrecedesSubcommand` method.

Reply:
I might not be understanding this part of the code correctly. The RootSubCommandResult method just returns _a_ valid name for a subcommand it finds? It looked to me like it returned _the_ subcommand the user actually used, in which case searching for exactly that subcommand makes sense.

Is there a list of aliases somewhere I can reference?

Reply:
Oh, you are likely right if `RootSubCommandResult` just returns what the user provided. I was looking at the construction of `CliCommand` itself as 'sub-commands' are also just `CliCommand`. And that class contains a list of aliases.

I'm not anything close to an expert at this stuff as I haven't written any CLI commands myself. I've only read some of the code, some of the docs, and fixed a few bugs so far. So, kinda a "blind leading the blind" situation here. 😅